src/arrayMethodJoin.js

```diff
@@ -5,8 +5,18 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let string = '';
+
+    if (separator !== undefined) {
+      string = this.toString().replace(/,/g, separator);
+
+      return (string);
+    } else {
+      string = this.toString();
+
+      return (string);
```

Comment:
You mustn't add brackets to var next to `return`
```suggestion
      return string;
```